astroid/tree/node_classes.py

```diff
@@ -129,54 +147,25 @@ class Name(base.LookupMixIn, base.NodeNG):
     def __init__(self, name=None, lineno=None, col_offset=None, parent=None):
         self.name = name
         super(Name, self).__init__(lineno, col_offset, parent)
-
+    
 
 @util.register_implementation(treeabc.Arguments)
 class Arguments(base.AssignTypeMixin, AssignedStmtsMixin, base.NodeNG):
     """class representing an Arguments node"""
-    if six.PY3:
-        # Python 3.4+ uses a different approach regarding annotations,
-        # each argument is a new class, _ast.arg, which exposes an
-        # 'annotation' attribute. In astroid though, arguments are exposed
-        # as is in the Arguments node and the only way to expose annotations
-        # is by using something similar with Python 3.3:
-        #  - we expose 'varargannotation' and 'kwargannotation' of annotations
-        #    of varargs and kwargs.
-        #  - we expose 'annotation', a list with annotations for
-        #    for each normal argument. If an argument doesn't have an
-        #    annotation, its value will be None.
-
-        _astroid_fields = ('args', 'defaults', 'kwonlyargs',
-                           'kw_defaults', 'annotations', 'kwonly_annotations',
-                           'varargannotation', 'kwargannotation')
-        varargannotation = None
-        kwargannotation = None
-    else:
-        _astroid_fields = ('args', 'defaults', 'kwonlyargs', 'kw_defaults')
-    _other_fields = ('vararg', 'kwarg')
 
-    def __init__(self, vararg=None, kwarg=None, parent=None):
+    _astroid_fields = ('args', 'vararg', 'kwarg', 'keyword_only', 'positional_only')
+
+    def __init__(self, parent=None):
+        # We don't want lineno and col_offset from the parent's __init__.
+        super(Arguments, self).__init__(parent=parent)
+
+    def postinit(self, args, vararg, kwarg, keyword_only, positional_only):
+        self.args = args
         self.vararg = vararg
         self.kwarg = kwarg
-        self.parent = parent
-        self.args = []
-        self.defaults = []
-        self.kwonlyargs = []
-        self.kw_defaults = []
-        self.annotations = []
-        self.kwonly_annotations = []
-
-    def postinit(self, args, defaults, kwonlyargs, kw_defaults,
-                 annotations, kwonly_annotations, varargannotation=None,
-                 kwargannotation=None):
-        self.args = args
-        self.defaults = defaults
-        self.kwonlyargs = kwonlyargs
-        self.kw_defaults = kw_defaults
-        self.annotations = annotations
-        self.varargannotation = varargannotation
-        self.kwargannotation = kwargannotation
-        self.kwonly_annotations = kwonly_annotations
+        self.keyword_only = keyword_only
+        self.positional_only = positional_only
+        self.positional_and_keyword = self.args + self.positional_only
```

Comment:
This should be a read-only property to avoid the possibility of these attributes getting unsynchronized from one another.  If we ever make the nodes immutable, that would also fix the problem.